### PR TITLE
Feature/context

### PR DESCRIPTION
**Pull request by JakobJK**
Adding a new feature - Contexts

Basically, it means that things that are going on in model_A, will not interfere with things going on in model_B. (So you can have the same name in two different models, but still pass the duplicated names check for example).

image

### CODE REVIEW

File: modelChecker/modelChecker_commands.py [hunk 1]
```diff
@@ -9,11 +9,19 @@ def trailingNumbers(nodes, _):
...
+        shortName = node.split('|')[-1]
+        if shortName in duplicatedNames:
+            duplicatedNames[shortName].append(node)
+            store.add(shortName)
+        else:
+            duplicatedNames[shortName] = [node]
+    output = []
+    for name in store:
+        output.extend(duplicatedNames[name])
+    return output
```
Comment (BigRoy) [inline modelChecker/modelChecker_commands.py]: Can also be written as:
```python
nodes_by_short_name = defaultdict(list)
for node in nodes:
    short_name = node.rsplit("|", 1)[-1]
    nodes_by_short_name[short_name].append(node)

invalid = []
for short_name, short_name_nodes in nodes_by_short_name.items():
    if len(short_name_nodes) > 1:
        invalid.extend(short_name_nodes)
return invalid
```

File: modelChecker/modelChecker_commands.py [hunk 2]
```diff
@@ -129,26 +136,21 @@ def zeroLengthEdges(_, SLMesh):
...
     selfPenetratingUVs = []
     for node in transformNodes:
-        shape = cmds.listRelatives(node, shapes=True, fullPath=True)
-        convertToFaces = cmds.ls(
-            cmds.polyListComponentConversion(shape, tf=True), fl=True)
-        overlapping = (cmds.polyUVOverlap(convertToFaces, oc=True))
-        if overlapping:
-            for node in overlapping:
-                selfPenetratingUVs.append(node)
+        shape = cmds.listRelatives(node, shapes=True, fullPath=True, type="mesh")
```
Comment (BigRoy) [inline modelChecker/modelChecker_commands.py]: You might want to exclude intermediateObjects by adding the `noIntermediate=True` argument.

File: modelChecker/modelChecker_commands.py [hunk 3]
```diff
@@ -129,26 +136,21 @@ def zeroLengthEdges(_, SLMesh):
...
+        faces = cmds.polyListComponentConversion(shape, tf=True)
+        if overlapping := cmds.polyUVOverlap(faces, oc=True):
```
Comment (BigRoy) [inline modelChecker/modelChecker_commands.py]: I wonder if this also works.
```suggestion
        if overlapping := cmds.polyUVOverlap("{}.f[*]".format(shape), oc=True):
```

Comment (JakobJK) [reply modelChecker/modelChecker_commands.py]: I like it! I'll give it a shot. Great suggestion!

File: modelChecker/modelChecker_commands.py (hunk 2 above)
Comment (JakobJK) [reply modelChecker/modelChecker_commands.py]: That makes sense to me!

File: modelChecker/modelChecker_commands.py (hunk 1 above)
Comment (JakobJK) [reply modelChecker/modelChecker_commands.py]: Nice! Looks a lot neater!

**Pull request merged**